packages/notarization/manager_test.go

```diff
@@ -9,195 +9,847 @@ import (
+		m.pendingConflictsCounters[epoch.Index(i)] = uint64(i)
 		err := m.epochCommitmentFactory.insertTangleLeaf(epoch.Index(i), tangle.EmptyMessageID)
 		require.NoError(t, err)
 	}
 
 	commitment, err := m.GetLatestEC()
 	assert.NoError(t, err)
 	// only epoch 0 has pbc = 0
-	assert.Equal(t, epoch.Index(0), commitment.EI)
+	assert.Equal(t, epoch.Index(0), commitment.EI())
 
-	m.pendingConflictsCount[4] = 0
+	m.pendingConflictsCounters[4] = 0
 	commitment, err = m.GetLatestEC()
 	assert.NoError(t, err)
 	// epoch 4 has pbc = 0 but is not old enough
-	assert.Equal(t, epoch.Index(0), commitment.EI)
+	assert.Equal(t, epoch.Index(0), commitment.EI())
 
-	m.pendingConflictsCount[2] = 0
+	m.pendingConflictsCounters[1] = 0
+	m.pendingConflictsCounters[2] = 0
 	commitment, err = m.GetLatestEC()
 	assert.NoError(t, err)
 	// epoch 2 has pbc=0 and is old enough
-	assert.Equal(t, epoch.Index(2), commitment.EI)
+	assert.Equal(t, epoch.Index(2), commitment.EI())
 }
 
 func TestManager_UpdateTangleTree(t *testing.T) {
+	epochInterval := 1 * time.Second
+
+	nodes := make(map[string]*identity.Identity)
+	for _, node := range []string{"A", "B", "C", "D", "E"} {
+		nodes[node] = identity.GenerateIdentity()
+	}
+
+	var weightProvider *tangle.CManaWeightProvider
+	manaRetrieverMock := func() map[identity.ID]float64 {
+		for _, node := range nodes {
+			weightProvider.Update(time.Now(), node.ID())
+		}
+		return map[identity.ID]float64{
+			nodes["A"].ID(): 30,
+			nodes["B"].ID(): 20,
+			nodes["C"].ID(): 25,
+			nodes["D"].ID(): 25,
+		}
+	}
+	weightProvider = tangle.NewCManaWeightProvider(manaRetrieverMock, time.Now)
+
+	testFramework, eventHandlerMock, notarizationMgr := setupFramework(t, epochInterval, tangle.ApprovalWeights(weightProvider), tangle.WithConflictDAGOptions(conflictdag.WithMergeToMaster(false)))
+
+	// Message1, issuing time epoch 1
+	{
+		time.Sleep(epochInterval)
+		fmt.Println("message 1")
+
+		ecRecord, _, err := testFramework.LatestCommitment()
+		require.NoError(t, err)
+		testFramework.CreateMessage("Message1", tangle.WithStrongParents("Genesis"), tangle.WithIssuer(nodes["A"].PublicKey()), tangle.WithECRecord(ecRecord))
+		testFramework.IssueMessages("Message1").WaitUntilAllTasksProcessed()
+
+		msg := testFramework.Message("Message1")
+		assert.Equal(t, epoch.Index(0), msg.EI())
+	}
+	// Message2, issuing time epoch 2
+	{
+		time.Sleep(epochInterval)
+		fmt.Println("message 2")
+
+		ecRecord, _, err := testFramework.LatestCommitment()
+		require.NoError(t, err)
+		testFramework.CreateMessage("Message2", tangle.WithStrongParents("Message1"), tangle.WithIssuer(nodes["B"].PublicKey()), tangle.WithECRecord(ecRecord))
+		testFramework.IssueMessages("Message2").WaitUntilAllTasksProcessed()
+
+		msg := testFramework.Message("Message2")
+		assert.Equal(t, epoch.Index(0), msg.EI())
+	}
+
+	assertExistenceOfBlock(t, testFramework, notarizationMgr, map[string]bool{
+		"Message1": true,
+	})
+
+	// Message3, issuing time epoch 3
```

Comment:
We could start from here also checking if the reference to the `prevEC` is correct